towerlib/towerlib.py

```diff
@@ -2018,3 +2031,178 @@ def get_schedule_by_name(self, name):
 
         """
         return next(self.schedules.filter({'name__iexact': name}), None)
+
+    def update_all_organization_projects(self, organization_name):
+        """Update all the projects in ansible tower for a given organization.
+
+        Args:
+            organization_name: The name of the organization.
+
+        """
+        organization = self.get_organization_by_name(organization_name)
+        if not organization:
+            raise InvalidOrganization(organization_name)
+        outputs = []
+        for project in organization.projects:
+            output = project.update()
+            if not output:
+                self._logger.error(f'{project.name} failed to update')
+            outputs.append(output)
+        return all(outputs)
+
+    def update_project_by_id(self, project_id):
+        """Update the ansible tower project with given project id.
+
+        Args:
+            project_id: The id of the project, which is to be updated.
+
+        Returns:
+            list: List of response of api request as json on success, None otherwise.
+
+        """
+        project = self.get_project_by_id(project_id)
+        if not project:
+            raise InvalidProject(project_id)
+        return project.update()
+
+    def update_organization_project_by_name(self, organization_name, project_name):
+        """Update the ansible tower project with given project name.
+
+        Args:
+            organization_name: The name of the organization.
+            project_name: The name of the project, which is to be updated.
+
+        Returns:
+            API Response (dict): dict of response of api request as json on success, None otherwise.
+
+        """
+
+        organization = self.get_organization_by_name(organization_name)
```

Comment:
Here you should use the `get_organization_project_by_name` call to save a network call. The way it is currently implemented does the same call to retrieve the organiazation twice which is wasteful. Basically you don't need lines 2080 to 2082.